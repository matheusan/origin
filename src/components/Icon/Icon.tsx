import * as React from 'react';

import academy from '../../assets/icons/academy.svg';
import arrow from '../../assets/icons/arrow.svg';
import dollar from '../../assets/icons/dollar.svg';
import house from '../../assets/icons/house.svg';
import loading from '../../assets/icons/loading.svg';
import logo from '../../assets/icons/logo.svg';
import tip from '../../assets/icons/tip.svg';

const icons = {
  academy = academy,
  arrow = arrow,
  dollar = dollar,
  house = house,
  loading = loading,
  logo = logo,
  tip = tip
};

export type Icons = keyof typeof icons;
export interface IconProps {
  name: Icons;
  width?: string;
  height?: string;
}

const Icon: React.SFC<IconProps & React.HTMLAttributes> = ({
  name,
  width = '',
  height = '',
  ...otherProps
}): JSX.Element | null => {
  if (!icons[name]) {
    return;
  }

  return (
    <img src={icons[name]} width={width} height={height} {...otherProps} />
  );
};

export default Icon;
