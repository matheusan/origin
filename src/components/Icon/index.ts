import Icon from './Icon';
import { Icons } from './Icon';
export { Icons };
export default Icon;
