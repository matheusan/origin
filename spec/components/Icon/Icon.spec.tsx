import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Icon from '../../../src/components/Icon';

describe('Icon', () => {
  let wrapper: ShallowWrapper;

  it('renders', () => {
    wrapper = shallow(<Icon name="academy" />);
    expect(wrapper).toMatchSnapshot();
  });

  [
    'academy',
    'arrow',
    'dollar',
    'house',
    'loading',
    'logo',
    'tip',
    'piggy',
    'vacation',
    'car',
    'wedding',
    'vault',
    'baby'
  ].forEach(icon => {
    it(`should render ${icon} icon`, () => {
      wrapper = shallow(<Icon name={icon} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('renders fragment if invalid icon name is given', () => {
    wrapper = shallow(<Icon name="noop" />);
    expect(wrapper.find('Fragment').length).toBe(1);
  });

  it('renders fragment if no icon name is given', () => {
    wrapper = shallow(<Icon />);
    expect(wrapper.find('Fragment').length).toBe(1);
  });
});
