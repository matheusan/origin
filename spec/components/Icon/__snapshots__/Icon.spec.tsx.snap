// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render academy icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render arrow icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render baby icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render car icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render dollar icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render house icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render loading icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render logo icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render piggy icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render tip icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render vacation icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render vault icon 1`] = `
<img
  height=""
  width=""
/>
`;

exports[`Icon should render wedding icon 1`] = `
<img
  height=""
  width=""
/>
`;
